### PR TITLE
annotate the auto-generated spidermMultusConfig by init-pod

### PR DESCRIPTION
**Pull request by cyclinder**
---
name: Pull request
about: Tell us about your contribution
labels: ["pr/release/none-required"]
---

---
## Thanks for contributing !

Before submitting a pull request, make sure you read about our Contribution notice here: <https://spidernet-io.github.io/spiderpool/contributing/development/pullrequest/#need-review>

**What this PR does / why we need it**:

it's for production. We should annotate some known annotations to the auto-generated spidermMultusConfig.

For examples:

```
  annotations:
    v1.multus-underlay-cni.io/coexist-types: '["macvlan_overlay","sriov_overlay"]'
    v1.multus-underlay-cni.io/default-cni: "true"
    v1.multus-underlay-cni.io/instance-type: default
    v1.multus-underlay-cni.io/vlanId: "0"
  labels:
    v1.multus-underlay-cni.io/instance-status: enable
```

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**make sure your commit is signed off**

### CODE REVIEW

**Comment by cyclinder** [comment]
```
• [FAILED] [0.063 seconds]
test annotation [BeforeEach] [annotation]
/home/runner/work/spiderpool/spiderpool/test/e2e/annotation/annotation_test.go:30
  invalid annotations table
  /home/runner/work/spiderpool/spiderpool/test/e2e/annotation/annotation_test.go:94
    fail to run a pod with non-existed ippool v4、v6 values [A00003]
    /home/runner/work/spiderpool/spiderpool/test/e2e/annotation/annotation_test.go:121

  Timeline >>
  > Enter [BeforeEach] test annotation - /home/runner/work/spiderpool/spiderpool/test/e2e/annotation/annotation_test.go:30 @ 07/25/23 03:37:04.95
  Create v4 subnet  and v4 pool v4pool-374-951752252 
  [FAILED] Unexpected error:
      <*errors.StatusError | 0xc0004e8d20>: 
      admission webhook "spidersubnet.spiderpool.spidernet.io" denied the request: SpiderSubnet.spiderpool.spidernet.io "v4-ss-374-951797853" is invalid: spec.subnet: Invalid value: "10.217.210.0/24": overlap with Subnet v4-ss-374-846760345 which 'spec.subnet' is 10.217.210.0/24
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "admission webhook \"spidersubnet.spiderpool.spidernet.io\" denied the request: SpiderSubnet.spiderpool.spidernet.io \"v4-ss-374-951797853\" is invalid: spec.subnet: Invalid value: \"10.217.210.0/24\": overlap with Subnet v4-ss-374-846760345 which 'spec.subnet' is 10.217.210.0/24",
              Reason: "Invalid",
              Details: {
                  Name: "v4-ss-374-951797853",
                  Group: "spiderpool.spidernet.io",
                  Kind: "SpiderSubnet",
                  UID: "",
                  Causes: [
                      {
                          Type: "FieldValueInvalid",
                          Message: "Invalid value: \"10.217.210.0/24\": overlap with Subnet v4-ss-374-846760345 which 'spec.subnet' is 10.217.210.0/24",
                          Field: "spec.subnet",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 422,
          },
      }
  occurred
  In [BeforeEach] at: /home/runner/work/spiderpool/spiderpool/test/e2e/annotation/annotation_test.go:40 @ 07/25/23 03:37:05.013
  < Exit [BeforeEach] test annotation - /home/runner/work/spiderpool/spiderpool/test/e2e/annotation/annotation_test.go:30 @ 07/25/23 03:37:05.013 (62ms)
  << Timeline
------------------------------
P [PENDING]
test annotation [annotation]
```

File: cmd/spiderpool-init/cmd/config.go [hunk 1]
```diff
@@ -330,10 +330,23 @@ func getMultusCniConfig(cniName, cniType string) *spiderpoolv2beta1.SpiderMultus
...
+			},
 		},
 		Spec: spiderpoolv2beta1.MultusCNIConfigSpec{
 			CniType:           spiderpoolv2beta1.CniType(cniType),
 			EnableCoordinator: pointer.Bool(false),
 		},
 	}
 }
+
+func getDefaultAnnotations() (annotations map[string]string) {
```
Comment (weizhoublue) [inline cmd/spiderpool-init/cmd/config.go:344]: 前端是直接获取 cr 还是通过 spidernet ？
如果是通过 spidernet ，这些东西是否 让 spidernet 加上 返给 前端，毕竟 这些是 产品定制的，和开源是两个维度

Comment (cyclinder) [reply cmd/spiderpool-init/cmd/config.go:344]: 通过spidernet获取。但每次获取时都加一下是不是也太冗余了，这只是一些annotations，而且本来自动生成spidermultusconfig 也是为了产品化

Comment (weizhoublue) [reply cmd/spiderpool-init/cmd/config.go:344]: 冗余 --反正是机器，不存在冗余一说

一方面，这是开源， 不是产品，。怎么解释 这些 annotation
要是以后， 我担心 以后 变了 ，开源跟着变

我认为 在 spidernet 应该是最优的

Comment (cyclinder) [reply cmd/spiderpool-init/cmd/config.go:344]: got

**Pull request closed without merging**